erpnext/edi/doctype/code_list/code_list.py

```diff
@@ -78,3 +82,23 @@ def get_code_for(code_list: str, doctype: str, name: str) -> str | None:
 	).run()
 
 	return codes[0][0] if codes else None
+
+
+def get_doc_for(code_list: str, doctype: str, code: str) -> str | None:
+	"""Return the record name for a given common code"""
+	CommonCode = frappe.qb.DocType("Common Code")
+	DynamicLink = frappe.qb.DocType("Dynamic Link")
+
+	docnames = (
+		frappe.qb.from_(CommonCode)
+		.join(DynamicLink)
+		.on((CommonCode.name == DynamicLink.parent) & (DynamicLink.parenttype == "Common Code"))
+		.select(DynamicLink.link_name)
+		.where(
+			(DynamicLink.link_doctype == doctype)
+			& (CommonCode.common_code == code)
+			& (CommonCode.code_list == code_list)
+		)
+	).run()
+
+	return docnames[0][0] if docnames else None
```

Comment:
So, I guess this will break in a subset of valid cases, namely when we can't assume `common_code` to be a unique column, right?

I think we should avoid to offer unbound behaviour. For example, we could return a UserError condition if we detect more than one row in the query result. This would even relax the unique constraint required by this getter to `(codelist, doctype, code)` without precluding that the underlying dataset can't exhibit such records.